internal/gardener/gardener.go

```diff
@@ -153,6 +153,16 @@ func (g *gardenerProvisioner) validate(cluster *types.Cluster, provider *types.P
 	} else {
 		errMessage += fmt.Sprintf(errs.CannotBeEmpty, "Provider.CustomConfigurations['target_provider']")
 	}
+	if v, ok := provider.CustomConfigurations["target_profile"]; ok {
```

Comment:
Do we really need this configuration? As I can see, the possible profiles are: GCP, Azure and AWS. I guess they always match the `target_provider` (e.g. if `target_provider` is `gcp`, the `target_profile` will always be `gcpProfile`).
If this is the case, why do we need to explicitly provide the `target_profile` if we always can infer it from the `target_provider`?
Is there any case where `target_profile` and `target_provider` are not the same?

Reply:
removed it from required variables and inferred the value from the provider